### PR TITLE
Fix typos and small cleanup.

### PR DESCRIPTION
**Pull request by dereuromark**
I didnt think that cake in v5 had that many typos still in the code.
The IDE found those issues mainly.

### CODE REVIEW

**Comment by dereuromark** [comment]
The PHP 8.4 fails seem unrelated.

File: src/Datasource/EntityTrait.php [hunk 1]
```diff
@@ -327,9 +327,7 @@ public function &get(string $field): mixed
 
         $method = static::_accessor($field, 'get');
         if ($method) {
-            $result = $this->{$method}($value);
-
-            return $result;
+            return $this->{$method}($value);
```
Comment (ADmad) [inline src/Datasource/EntityTrait.php]: This change is causing
```
Notice: Only variable references should be returned by reference in /home/runner/work/cakephp/cakephp/src/Datasource/EntityTrait.php on line 330
```

Comment (dereuromark) [reply src/Datasource/EntityTrait.php]: Is this correct? Looks weird that this variable needs to be existing just for the return to work^^
Almost like a PHP bug of sorts.
I can revert this.

File: src/Console/Arguments.php [hunk 2]
```diff
@@ -164,6 +164,7 @@ public function getOption(string $name): string|bool|null
     /**
      * Get a boolean option's value or null if not set.
      *
+     * @param string $name Name
```
Comment (ADmad) [inline src/Console/Arguments.php:167]: I would go the opposite way and remove such redundant annotations everywhere :)

Comment (dereuromark) [reply src/Console/Arguments.php:167]: Right now it is consistent to have them all matching.
It also increases readability (as the order relates to where to find them).
If we would drop them randomly, things go boom soon.
As you can see in other places, they might also get mixed up in order when adding.

So overall I am not in favor of removing them.

**Pull request merged**